WooCommerce/WooCommerceTests/POS/Models/PointOfSaleAggregateModelTests.swift

```diff
@@ -428,6 +346,63 @@ struct PointOfSaleAggregateModelTests {
             #expect(sut.orderStage == .building)
         }
 
+        @Test func checkOut_when_reader_connects_collectPayment_called() async throws {
+            // Given
+            cardPresentPaymentService.connectedReader = nil
+
+            orderController.orderStateToReturn = makeLoadedOrderState()
+            await sut.checkOut()
+            cardPresentPaymentService.collectPaymentWasCalled = false
+
+            // When
+            // `await confirmation` callback only waits until this completes, not until some timeout.
+            // Since this is synchonous but triggers async combine behaviour, we can't use that approach.
+            cardPresentPaymentService.connectedReader = .init(name: "Test reader", batteryLevel: 0.7)
+
+            // Then
+            let timeout = ContinuousClock.now + .seconds(2)
+
+            while cardPresentPaymentService.collectPaymentWasCalled != true {
+                try await Task.sleep(for: .milliseconds(1))
+                try #require(.now < timeout)
+            }
```

Comment:
I'm guessing this is the replacement of the XCTest `waitUntil` helper in Swift Testing? Maybe we can make a helper like that in TestKit so that it's easier to discover.

Reply:
Yes, it's not that nice but I'm not sure whether we want to encourage it, actually. It would be better to move towards more use of async code which doesn't need this kind of waiting around!